execution_engine/src/storage/trie_store/db.rs

```diff
@@ -2,76 +2,48 @@
+    MultiThreaded, Options,
+};
+
+use casper_types::{bytesrepr, bytesrepr::Bytes, Key, StoredValue};
 
 use casper_hashing::Digest;
 
 use crate::storage::{
     error,
     global_state::CommitError,
-    store::Store,
-    transaction_source::{
-        db::{LmdbEnvironment, RocksDb, RocksDbStore},
-        Readable, Transaction, TransactionSource, Writable,
-    },
+    store::{ErrorSource, Readable, Store, Writable},
     trie::Trie,
-    trie_store::{self, TrieStore},
+    trie_store::TrieStore,
 };
 
-/// An LMDB-backed trie store.
-///
-/// Wraps [`lmdb::Database`].
-#[derive(Debug, Clone)]
-pub struct LmdbTrieStore {
-    db: Database,
-}
-
-impl LmdbTrieStore {
-    /// Constructor for new `LmdbTrieStore`.
-    pub fn new(
-        env: &LmdbEnvironment,
-        maybe_name: Option<&str>,
-        flags: DatabaseFlags,
-    ) -> Result<Self, error::Error> {
-        let name = Self::name(maybe_name);
-        let db = env.env().create_db(Some(&name), flags)?;
-        Ok(LmdbTrieStore { db })
-    }
-
-    /// Constructor for `LmdbTrieStore` which opens an existing lmdb store file.
-    pub fn open(env: &LmdbEnvironment, maybe_name: Option<&str>) -> Result<Self, error::Error> {
-        let name = Self::name(maybe_name);
-        let db = env.env().open_db(Some(&name))?;
-        Ok(LmdbTrieStore { db })
-    }
-
-    fn name(maybe_name: Option<&str>) -> String {
-        maybe_name
-            .map(|name| format!("{}-{}", trie_store::NAME, name))
-            .unwrap_or_else(|| String::from(trie_store::NAME))
-    }
-
-    /// Get a handle to the underlying database.
-    pub fn get_db(&self) -> Database {
-        self.db
-    }
-}
+/// Relative location (to storage) where rocksdb data will be stored.
+pub const ROCKS_DB_DATA_DIR: &str = "rocksdb-data";
 
-impl<K, V> Store<Digest, Trie<K, V>> for LmdbTrieStore {
-    type Error = error::Error;
+const ROCKS_DB_BLOCK_SIZE_BYTES: usize = 256 * 1024;
+const ROCKS_DB_COMPRESSION_TYPE: rocksdb::DBCompressionType = rocksdb::DBCompressionType::Zstd;
+const ROCKS_DB_COMPACTION_STYLE: rocksdb::DBCompactionStyle = rocksdb::DBCompactionStyle::Level;
+const ROCKS_DB_ZSTD_MAX_DICT_BYTES: i32 = 256 * 1024;
+const ROCKS_DB_MAX_LEVEL_FILE_SIZE_BYTES: u64 = 512 * 1024 * 1024;
+const ROCKS_DB_MAX_OPEN_FILES: i32 = 768;
+const ROCKS_DB_ZSTD_COMPRESSION_LEVEL: i32 = 3; // Default compression level
+const ROCKS_DB_ZSTD_STRATEGY: i32 = 4; // 4: Lazy
+const ROCKS_DB_WINDOW_BITS: i32 = -14;
 
-    type Handle = Database;
+/// Column family name for the v1 trie data store.
+const TRIE_V1_COLUMN_FAMILY: &str = "trie_v1_column";
 
-    fn handle(&self) -> Self::Handle {
-        self.db
-    }
-}
+/// Column family name for tracking progress of data migration.
+const LMDB_MIGRATED_STATE_ROOTS_COLUMN_FAMILY: &str = "lmdb_migrated_state_roots_column";
 
-impl<K, V> TrieStore<K, V> for LmdbTrieStore {}
+/// Working set column famliy.
+const WORKING_SET_COLUMN_FAMILY: &str = "working_set";
```

Comment:
There's no working set yet in this branch?

Reply:
Good catch. Fixed in 213bf27a50f00424def44d1557567f3f3d184e21